pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -475,6 +530,86 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *api.Ser
 		status.Ingress = append(status.Ingress, api.LoadBalancerIngress{IP: floatIP.FloatingIP})
 	}
 
+	if lbaas.opts.ManageSecurityGroups {
+
+		lbSecGroupCreateOpts := groups.CreateOpts{
+			Name:        getSecurityGroupName(clusterName, apiService),
+			Description: fmt.Sprintf("Securty Group for %v Service LoadBalancer", apiService.Name),
+		}
+
+		lbSecGroup, err := groups.Create(lbaas.network, lbSecGroupCreateOpts).Extract()
+
+		if err != nil {
+			return nil, err
+		}
+
+		for _, port := range ports {
+
+			for _, sourceRange := range sourceRanges.StringSlice() {
+				lbSecGroupRuleCreateOpts := rules.CreateOpts{
+					Direction:      "ingress",
+					PortRangeMax:   int(port.Port),
+					PortRangeMin:   int(port.Port),
+					Protocol:       strings.ToLower(string(port.Protocol)),
+					RemoteIPPrefix: string(sourceRange),
+					SecGroupID:     lbSecGroup.ID,
+					EtherType:      "IPv4",
+				}
+
+				_, err := rules.Create(lbaas.network, lbSecGroupRuleCreateOpts).Extract()
+
+				if err != nil {
+					return nil, err
+				}
+
+				lbaas.createNodeSecurityGroup(int(port.Port), string(port.Protocol), lbSecGroup.ID)
+			}
+
+		}
+
+		var port neutron_ports.Port
+
+		// Get the port ID
+		portID, err := getPortIDByIP(lbaas.network, loadbalancer.VipAddress)
+		if err != nil {
+			// cleanup what was created so far
+			_ = lbaas.EnsureLoadBalancerDeleted(clusterName, apiService)
+			return nil, err
+		}
+
+		opts := neutron_ports.ListOpts{ID: portID}
+
+		// Get the full port detail
+		err = neutron_ports.List(lbaas.network, opts).EachPage(func(page pagination.Page) (bool, error) {
+			prts, err := neutron_ports.ExtractPorts(page)
+			if err != nil {
+				glog.Errorf("Failed to extract ports: %v", err)
+				return false, err
+			}
+			// Getting with an ID filter should have one result, and it should match the ID
+			if len(prts) != 1 || prts[0].ID != portID {
+				glog.Errorf("Failed to find port")
+				return false, err
+			}
+			port = prts[0]
+			return true, nil
+		})
+		if err != nil {
+			glog.Errorf("Error occured getting port: %s", portID)
+			return nil, err
+		}
+
+		update_opts := neutron_ports.UpdateOpts{SecurityGroups: append(port.SecurityGroups, lbSecGroup.ID)}
+
+		res := neutron_ports.Update(lbaas.network, port.ID, update_opts)
+
+		if res.Err != nil {
+			glog.Errorf("Error occured updating port: %s", portID)
+			return nil, err
```

Comment:
Wrong error, should be:

```
return nil, res.Err
```

Alternatively, elsewhere in the openstack plugin we've used something more like this:
```
_, err := neutron_ports.Update(...)
if err != nil {
   glog.Errorf("Error ...")
   return nil, err
}
```